### PR TITLE
Wizard recipe: WORLD-v0.1.0

### PR DESCRIPTION
**Pull request by jkrumbiegel**
This pull request contains a new build recipe I built using the BinaryBuilder.jl wizard:

* Package name: WORLD
* Version: v0.1.0

@staticfloat please review and merge.

### CODE REVIEW

**Comment by jkrumbiegel** [comment]
good, this is the state as I had it locally. Linux builds go through, Apple and Windows fail.

Even if those two built, I'm not sure if I could use the library as intended, as I'm not sure if the symbols that I need will be accessible through ccall, as there are no `extern C` declarations in the code, and I assumed those were needed.

File: W/WORLD/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    GitSource("https://github.com/mmorise/World.git", "1f8815f7a85e344ea9f98ca6b1383578a161d4e2")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd World/
+ls
+mkdir -p ${libdir}
+$CXX -O1 -Wall -shared -fPIC -o ${libdir}/libworld.${dlext} -Isrc src/*.cpp
```
Comment (giordano) [inline W/WORLD/build_tarballs.jl:19]: Why `-O1`? Usually people want performance and would want to use higher optimisation levels

**Comment by giordano** [comment]
> I'm not sure if the symbols that I need will be accessible through ccall

What functions you want to ccall?

**Comment by giordano** [comment]
BTW, if you want to denote a PR as WIP and not merge it, please mark as draft instead of adding useless text to the title :wink:

File: W/WORLD/build_tarballs.jl (hunk 1 above)
Comment (jkrumbiegel) [reply W/WORLD/build_tarballs.jl:19]: good point, I copied that from the original makefile where it says `CXXFLAGS = -O1 -Wall -fPIC $(DEBUG_FLAG)`. Would you change it to O2 or even O3?

**Comment by jkrumbiegel** [comment]
> BTW, if you want to denote a PR as WIP and not merge it, please mark as draft instead of adding useless text to the title 😉

got it, never used a draft PR before but saw other people do the title thing :)

**Comment by jkrumbiegel** [comment]
> What functions you want to ccall?

I just found out that all relevant functions are actually marked `extern C`, but it's not written out. They use a macro `define WORLD_BEGIN_C_DECLS extern "C" {`, that's why I didn't recognize it... so actually this should be almost ready to go then

File: W/WORLD/build_tarballs.jl (hunk 1 above)
Comment (giordano) [reply W/WORLD/build_tarballs.jl:19]: Yes, probably 3 is a good one

File: W/WORLD/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "WORLD"
+version = v"0.1.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/mmorise/World.git", "1f8815f7a85e344ea9f98ca6b1383578a161d4e2")
```
Comment (giordano) [inline W/WORLD/build_tarballs.jl:10]:
```suggestion
    GitSource("https://github.com/mmorise/World.git", "1f8815f7a85e344ea9f98ca6b1383578a161d4e2"),
```

**Pull request merged**